app/controllers/talks_controller.rb

```diff
@@ -38,7 +38,18 @@ def destroy
 
   def upvote
     @talk = Talk.find(params[:id])
-    @talk.votes.create
+    @newvote = true
+
+    # Determines whether or not a user has upvoted the current topic
```

Comment:
I think we can use a `where` clause to tighten this up.

```
existing_vote = @talk.votes.where(:user_id => current_user_id)
```

That will fetch the existing vote the user did (or a `nil` value).  Then existing vote we can quickly check if it's a nil value.

```
@talk.votes.create(:user_id => current_user.id) unless existing_vote
```